### PR TITLE
BugFix: make sure pickup_time is not null when setting the status to picked up

### PR DESCRIPTION
**Pull request by ncounter**
## What does this PR change?

BugFix: backport of https://github.com/SUSE/spacewalk/pull/16254

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/redhat/rhn/domain/action/ActionFactory.java [hunk 1]
```diff
@@ -950,6 +950,23 @@ public static void removeInvalidResults(Action action) {
         }
     }
 
+    /**
+     * Update the status of several rhnServerAction rows identified by server and action IDs.
+     * @param actionIn associated action of rhnServerAction records
+     * @param serverIds server Ids for which action is scheduled
+     * @param status {@link ActionStatus} object that needs to be set
+     */
+    public static void updateServerActionsPickedUp(Action actionIn, List<Long> serverIds, ActionStatus status) {
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/domain/action/ActionFactory.java:959]: I do not see why this function must have `ActionStatus` as input parameter?
The only sense of this is to set it to PickedUp, so we can hardcode it.

I made a PR to change this.

**Review by mcalmer — CHANGES_REQUESTED** [review]

File: java/code/src/com/redhat/rhn/domain/action/ActionFactory.java (hunk 1 above)
Comment (ncounter) [reply java/code/src/com/redhat/rhn/domain/action/ActionFactory.java]: PR merged into this one, thank you

Comment (ncounter) [reply java/code/src/com/redhat/rhn/domain/action/ActionFactory.java]: I see you already implemented the change and it's pending in https://github.com/SUSE/spacewalk/pull/16285/commits . So I will backport it only to 4.2 then. Is that okay?

Comment (ncounter) [reply java/code/src/com/redhat/rhn/domain/action/ActionFactory.java]: https://github.com/SUSE/spacewalk/pull/16298

Comment (mcalmer) [reply java/code/src/com/redhat/rhn/domain/action/ActionFactory.java]: I have it already in https://github.com/SUSE/spacewalk/pull/16286. Let's do a race who merge first :-)

**Review by mcalmer — APPROVED** [review]

**Pull request merged**